dm-manager/pkg/device/device.go

```diff
@@ -301,20 +301,13 @@ func (dc *Controller) deactivateAMT(
 		err = errors.Errorf("%v", string(deactivateStatus.Body))
 		log.Err(err).
 			Msgf("expected to get 2XX, but got %v", deactivateStatus.StatusCode())
-		if deactivateStatus.StatusCode() == http.StatusNotFound {
```

Comment:
this is an happy path - you are removing the device and the device is not there. The job is already done.

Reply:
Handled the error case. Not required to add the device in MPS explicitly, after activation MPS will add it in DB. Tested activation and deactivation without adding explicitly and it is working.